packages/taiko-client/driver/preconf_blocks/api.go

```diff
@@ -48,6 +48,8 @@ type BuildPreconfBlockRequestBody struct {
 	ExecutableData    *ExecutableData `json:"executableData"`
 	EndOfSequencing   *bool           `json:"endOfSequencing"`
 	IsForcedInclusion *bool           `json:"isForcedInclusion"`
+	// Optional preconfirmation SignedCommitment from the sidecar, to be gossiped as-is.
+	Preconfirmation *p2p.SignedCommitment `json:"preconfirmation"`
```

Comment:
we should use an entirely new struct for this, and a new API, PreconfBlockV2 or whatever. Because all of the existing PreconfBlock data above, the `ExecutableData`, will be gone after the transition.

So, we should have an API that accepts the original struct, and a new API that accepts the PreconfirmationCommitment. Sidecars post-transition will call the new API, and use the old API until then.

After, we can delete the old API and struct.